src/ray/raylet/raylet_test.cc

```diff
@@ -0,0 +1,282 @@
+    if(ec != 0){
+      throw std::runtime_error("failed to start plasma store.");
+    };
+    return store_id;
+  }
+
+  void SetUp() {
+
+    // start store
+    std::string store_sock_1 = StartStore("1");
+    std::string store_sock_2 = StartStore("2");
+
+    // configure
+    std::unordered_map<string, double> static_resource_config;
+    static_resource_config = {{"num_cpus", 1}, {"num_gpus", 1}};
+    ray::ResourceSet resource_config(std::move(static_resource_config));
+
+    // start mock gcs
+    mock_gcs_client = shared_ptr<GcsClient>(new GcsClient());
+
+    // start first server
+    ray::OMConfig om_config_1;
+    om_config_1.store_socket_name = store_sock_1;
+    server1.reset(new Raylet(io_service,
+                                 std::string("hello1"),
+                                 resource_config,
+                                 om_config_1,
+                                 mock_gcs_client));
+
+    // start second server
+    ray::OMConfig om_config_2;
+    om_config_2.store_socket_name = store_sock_2;
+    server2.reset(new Raylet(io_service,
+                                 std::string("hello2"),
+                                 resource_config,
+                                 om_config_2,
+                                 mock_gcs_client));
+
+    // connect to stores.
+    ARROW_CHECK_OK(client1.Connect(store_sock_1, "", PLASMA_DEFAULT_RELEASE_DELAY));
+    ARROW_CHECK_OK(client2.Connect(store_sock_2, "", PLASMA_DEFAULT_RELEASE_DELAY));
+    this->StartLoop();
+  }
+
+  void TearDown() {
+    this->StopLoop();
+    arrow::Status client1_status = client1.Disconnect();
+    arrow::Status client2_status = client2.Disconnect();
+    ASSERT_TRUE(client1_status.ok() && client2_status.ok());
+
+    this->server1.reset();
+    this->server2.reset();
+
+    int s = system("killall plasma_store &");
+    ASSERT_TRUE(!s);
+
+    std::string cmd_str = test_executable.substr(0, test_executable.find_last_of("/"));
+    // cout << cmd_str << endl;
+    s = system(("rm " + cmd_str + "/hello1").c_str());
+    ASSERT_TRUE(!s);
+    s = system(("rm " + cmd_str + "/hello2").c_str());
+    ASSERT_TRUE(!s);
+  }
+
+  void Loop(){
+    io_service.run();
+  };
+
+  void StartLoop(){
+    p = std::thread(&TestRaylet::Loop, this);
+  };
+
+  void StopLoop(){
+    io_service.stop();
+    p.join();
+  }
+
+  ObjectID WriteDataToClient(plasma::PlasmaClient &client, int64_t data_size){
+    ObjectID object_id = ObjectID::from_random();
+    // cout << "ObjectID Created: " << object_id.hex().c_str() << endl;
+    uint8_t metadata[] = {5};
+    int64_t metadata_size = sizeof(metadata);
+    std::shared_ptr<Buffer> data;
+    ARROW_CHECK_OK(client.Create(object_id.to_plasma_id(), data_size, metadata, metadata_size, &data));
+    ARROW_CHECK_OK(client.Seal(object_id.to_plasma_id()));
+    return object_id;
+  }
+
+  void object_added_handler_1(const ObjectID &object_id){
+    cout << "Store 1 added: " << object_id.hex() << endl;
```

Comment:
presumably the printing was for debugging purposes, but most logging should be removed